diff --git a/csrc/py_itfs_cu/asm_gemm_a4w4.cu b/csrc/py_itfs_cu/asm_gemm_a4w4.cu
index c76d9107d..762ffdeaa 100644
--- a/csrc/py_itfs_cu/asm_gemm_a4w4.cu
+++ b/csrc/py_itfs_cu/asm_gemm_a4w4.cu
@@ -119,7 +119,7 @@ std::tuple<std::string, int> get_heuristic_kernel(int M,
                 std::vector<int> splitK_list =
                     (log2_k_split.has_value() && cfg.splitK)
                         ? std::vector<int>{1 << log2_k_split.value()}
-                        : (cfg.splitK ? std::vector<int>{2, 4, 8, 16} : std::vector<int>{1});
+                        : (cfg.splitK ? std::vector<int>{1, 2, 4, 8, 16} : std::vector<int>{1});
 
                 for(auto& splitK : splitK_list)
                 {
