diff --git a/aiter/jit/core.py b/aiter/jit/core.py
index 3f4777f6..7fa0ddf7 100644
--- a/aiter/jit/core.py
+++ b/aiter/jit/core.py
@@ -333,11 +333,10 @@ def build_module(
             flags_hip += ["-mllvm -amdgpu-coerce-illegal-types=1"]
         if get_gfx() == "gfx950" and int(os.getenv("AITER_FP4x2", "1")) > 0:
             flags_hip += ["-D__Float4_e2m1fn_x2"]
+        #import torch
 
-        import torch
-
-        if hasattr(torch, "float4_e2m1fn_x2"):
-            flags_hip += ["-DTORCH_Float4_e2m1fn_x2"]
+        #if hasattr(torch, "float4_e2m1fn_x2"):
+        flags_hip += ["-DTORCH_Float4_e2m1fn_x2"]
         flags_cc += flags_extra_cc
         flags_hip += flags_extra_hip
         archs = validate_and_update_archs()
diff --git a/aiter/jit/utils/cpp_extension.py b/aiter/jit/utils/cpp_extension.py
index e98b15ba..f9f0407e 100644
--- a/aiter/jit/utils/cpp_extension.py
+++ b/aiter/jit/utils/cpp_extension.py
@@ -1541,12 +1541,12 @@ def _write_ninja_file_to_build_library(
     # But we can't use this now because all aiter op based on torch
     # which means pybind11 related build flags must from torch now
     common_cflags = []
-    if torch_exclude and is_python_module:
-        import pybind11
+    #if torch_exclude and is_python_module:
+    #    import pybind11
 
-        extra_include_paths.append(pybind11.get_include())
-        common_cflags += [f"{x}" for x in _get_pybind11_abi_build_flags()]
-        common_cflags += [f"{x}" for x in _get_glibcxx_abi_build_flags()]
+     #   extra_include_paths.append(pybind11.get_include())
+     #   common_cflags += [f"{x}" for x in _get_pybind11_abi_build_flags()]
+     #   common_cflags += [f"{x}" for x in _get_glibcxx_abi_build_flags()]
 
     # sysconfig.get_path('include') gives us the location of Python.h
     # Explicitly specify 'posix_prefix' scheme on non-Windows platforms to workaround error on some MacOS
