diff --git a/csrc/ffi/gemm_fp4/gemm_fp4_ja.cu b/csrc/ffi/gemm_fp4/gemm_fp4_ja.cu
index 6a74821..ab6e44c 100644
--- a/csrc/ffi/gemm_fp4/gemm_fp4_ja.cu
+++ b/csrc/ffi/gemm_fp4/gemm_fp4_ja.cu
@@ -118,7 +118,7 @@ select_fp4_kernel(int M, int N, int K, int dev_id, CFG* cfgs) {
 
     if (cfg.tile_M != 128 || cfg.tile_N != 512 || (N % cfg.tile_N) == 0) {
       std::vector<int> splitK_list = cfg.splitK
-          ? std::vector<int>{2, 4, 8, 16}
+          ? std::vector<int>{1, 2, 4, 8, 16}
           : std::vector<int>{1};
 
       for (int splitK : splitK_list) {
